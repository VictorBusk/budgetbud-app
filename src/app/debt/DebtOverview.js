import React, {PureComponent} from 'react';
import {View, FlatList} from 'react-native';
import {connect} from 'react-redux';
import _ from 'lodash';
import {
    AppHeader,
    ConfirmDialog,
    Separator
} from "../../components/index";
import {button, text} from "../../style/index";
import {
    Container,
    Button,
    ListItem,
    Body,
    Right,
    Icon,
    Text,
    Spinner
} from 'native-base';
import I18n from "../../strings/i18n";
import {container, color} from "../../style/index";
import {showWarningToast} from "../../helpers/index";
import {debtSelected, deleteDebt, resetDebtError, resetDebtForm} from "./debt_actions";
import {getCategories} from "../category/category_actions";

class DebtOverview extends PureComponent {
    componentWillReceiveProps(nextProps) {
        if (nextProps.debtError) {
            showWarningToast(nextProps.debtError);
            this.props.resetDebtError();
        }
    }

    onCreateDebtPress = () => {
        this.props.resetDebtForm(() => {
            this.props.navigation.navigate('CreateDebt');
        });
    };

    deleteDebt = async () => {
        this.confirmDialog.dismissDialog();
        await this.props.deleteDebt(this.props.selectedDebt);
        await this.props.getCategories(this.props.budgetID);
    };

    onDebtSelect = (debt) => {
        this.props.debtSelected(debt);
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <ConfirmDialog
                    title={I18n.t('confirmDialogDeletionHeader')}
                    text={I18n.t('debtOverviewConfirmDialogBody')}
                    confirmCallback={() => this.deleteDebt()}
                    debtLoading={this.props.debtLoading}
                    ref={(confirmDialog) => {
                        this.confirmDialog = confirmDialog
                    }}
                />

                <Container>
                    <AppHeader headerText={I18n.t('debtOverviewHeader')}
                               showBackButton={true}
                               onLeftButtonPress={() => this.props.navigation.pop()}/>

                    <Container style={{flex: 4, justifyContent: 'center'}}>
                        {this.props.debtLoading ? (
                            <Spinner style={{
                                alignItems: 'center',
                                justifyContent: 'center'
                            }} color='#1c313a'/>) : (
                            <FlatList
                                data={this.props.debts}
                                renderItem={this.renderItem}
                            />
                        )}
                    </Container>

                    <Separator/>

                    <Button rounded
                            onPress={() => this.onCreateDebtPress()}
                            style={[button.defaultButton, color.button]}
                    >
                        <Text style={text.submitButtonText}>{I18n.t('debtOverviewCreateDebtButton')}</Text>
                    </Button>
                </Container>
            </Container>
        );
    }

    renderItem = ({item}) => {
        return (
            <ListItem>
                <Body>
                <Text style={color.text}>{item.name}</Text>
                <Text note>{item.totalAmount} {I18n.t('currency')}</Text>
                </Body>
                <Right>
                    <View style={{flexDirection: 'row'}}>
                        <Icon
                            onPress={() => {
                                this.onDebtSelect(item);
                                this.props.navigation.navigate('EditDebt');
                            }}
                            style={{marginRight: 7, fontSize: 30}}
                            name="md-create"/>
                        <Icon
                            onPress={() => {
                                this.onDebtSelect(item);
                                this.confirmDialog.showDialog()
                            }}
                            style={{marginHorizontal: 7, fontSize: 30}}
                            name="md-trash"/>
                    </View>
                </Right>
            </ListItem>
        );
    }
}

const mapStateToProps = (state) => {
    const budgetID = state.budget.budgetID;
    const {debtLoading, selectedDebt, debtError} = state.debt;

    const debts = _.map(state.debt.debts, (item, key) => {
        return {...item.debtData, debtID: item.id, key: key};
    });

    return {budgetID, debts, debtLoading, selectedDebt, debtError};
};

const mapDispatchToProps = {
    resetDebtForm, debtSelected, deleteDebt, getCategories, resetDebtError
};

export default connect(mapStateToProps, mapDispatchToProps)(DebtOverview);
