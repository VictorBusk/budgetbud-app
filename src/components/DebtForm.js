import React, {Component} from 'react';
import {
    View,
    FlatList,
    Keyboard,
    TouchableWithoutFeedback
} from 'react-native';
import {
    Container,
    Item,
    Input,
    Button,
    Text,
    Label,
    ListItem,
    CheckBox,
    Body,
    Right,
    Icon,
    Spinner
} from 'native-base';
import DatePicker from 'react-native-datepicker';
import {Separator} from '../components/';
import {
    button,
    container,
    color,
    text,
    input
} from "../style/";
import I18n from "../strings/i18n";
import {checkInputAmount} from "../helpers/validators";

export class DebtForm extends Component {
    onNameChange = (text) => {
        this.props.nameChanged(text);
    };

    onAmountChange = (amount) => {
        amount = amount.replace(/,/g, '.');
        if (checkInputAmount(amount))
            this.props.amountChanged(amount);
    };

    onExpirationDateChange = (text) => {
        this.props.expirationDateChanged(text);
    };

    onCheckBoxPress = ({categoryID}) => {
        let tmp = this.props.selectedCategories;

        if (tmp.includes(categoryID)) {
            tmp.splice(tmp.indexOf(categoryID), 1)
        } else {
            tmp.push(categoryID);
        }

        this.props.categoriesSelected(tmp);
    };

    render() {
        return (
            <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                <Container>
                    <View style={[container.defaultFormStyle, {paddingTop: 10}]}>
                        <Label style={[text.defaultText, color.text]}>{I18n.t('debtName')}</Label>
                        <Item rounded style={[input.inputField, color.input]}>
                            <Input
                                value={this.props.name}
                                onChangeText={this.onNameChange}
                                placeholder={I18n.t('debtNamePlaceholder')}
                                placeholderTextColor='#7F9BAA'
                                style={color.text}
                            />
                        </Item>
                    </View>

                    <View style={container.defaultFormStyle}>
                        <Label style={[text.defaultText, color.text]}>{I18n.t('debtAmount')}</Label>
                        <Item rounded style={[input.inputField, color.input]}>
                            <Input
                                value={String(this.props.amount)}
                                onChangeText={this.onAmountChange}
                                keyboardType="numeric"
                                placeholder={I18n.t('debtAmountPlaceholder')}
                                placeholderTextColor='#7F9BAA'
                                style={color.text}
                            />
                        </Item>
                    </View>

                    <View style={{marginVertical: 10}}>
                        <Separator/>
                    </View>

                    <View style={container.defaultFormStyle}>
                        <Label style={[text.defaultText, color.text]}>{I18n.t('debtExpirationDate')}</Label>
                        <DatePicker
                            date={this.props.expirationDate}
                            onDateChange={this.onExpirationDateChange}
                            format="DD-MM-YYYY"
                            minDate={new Date()}
                            style={[input.inputField, {width: '100%'}]}
                            cancelBtnText={I18n.t('datePickerCancelButton')}
                            confirmBtnText={I18n.t('datePickerOkButton')}
                            iconComponent={<Icon style={{color: '#295E80'}} name="md-calendar"/>}
                            customStyles={{
                                dateInput: {
                                    borderRadius: 100,
                                    borderColor: '#003755',
                                    backgroundColor: '#F6F7F9',
                                    borderWidth: 1,
                                    marginRight: 5
                                },
                                dateText: {
                                    color: '#003755',
                                    fontSize: 17
                                }
                            }}
                        />
                    </View>

                    <View style={{marginTop: 5, marginBottom: 10}}>
                        <Separator/>
                    </View>

                    <View style={{flex: 2, alignSelf: 'stretch'}}>
                        <View style={container.defaultFormStyle}>
                            <Label style={[text.defaultText, color.text]}>{I18n.t('debtCategories')}</Label>
                        </View>

                        {this.props.categoriesLoading ? (
                            <Container style={{justifyContent: 'center'}}>
                                <Spinner style={{
                                    alignItems: 'center',
                                    justifyContent: 'center'
                                }} color='#1c313a'/>
                            </Container>) : (
                            <FlatList
                                data={this.props.categoryItems}
                                renderItem={this.renderItem}
                                keyExtractor={item => item.categoryID}
                            />
                        )}
                    </View>

                    <View style={{marginTop: 5}}>
                        <Separator/>
                    </View>

                    <Button rounded
                            onPress={() => this.props.onContinuePress()}
                            style={[button.defaultButton, color.button]}
                    >
                        {this.props.debtLoading ? (
                            <Spinner color='#D0D0D0'/>) : (
                            <Text style={text.submitButtonText}>{I18n.t('debtContinueButton')}</Text>
                        )}
                    </Button>
                </Container>
            </TouchableWithoutFeedback>
        );
    }

    renderItem = ({item}) => {
        const checked = this.props.selectedCategories.includes(item.categoryID);

        return (
            <ListItem style={{paddingLeft: 9}}>
                <CheckBox
                    style={checked ? color.checkboxChecked : color.checkboxUnchecked}
                    checked={checked}
                    onPress={() => this.onCheckBoxPress(item)}
                />
                <Body>
                <Text style={color.text}>{item.name}</Text>
                </Body>
                <Right style={{paddingRight: 9}}>
                    <Text style={color.text}>{item.amount} {I18n.t('currency')}</Text>
                </Right>
            </ListItem>
        );
    }
}
