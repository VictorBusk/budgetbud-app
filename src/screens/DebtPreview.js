import React, {Component} from 'react';
import {Container, Content, Button, Text} from "native-base";
import {connect} from "react-redux";
import AppHeader from "../components/AppHeader";
import Separator from '../components/Separator';
import {createDebt} from "../actions";
import I18n from "../strings/i18n";

class DebtPreview extends Component {
    onSavePress = () => {
        this.props.createDebt(this.props, () => {
            this.props.navigation.pop(2);
        });
    };

    render() {
        return (
            <Container>
                <AppHeader headerText={I18n.t('debtPreviewHeader')}

                           showBackButton={true}
                           onLeftButtonPress={() => this.props.navigation.pop()}/>

                <Content style={{flex: 4}}>

                </Content>

                <Separator/>

                <Button rounded
                        onPress={() => this.onSavePress()}
                        style={styles.buttonStyle}
                >
                    <Text style={styles.itemStyle}>
                        {I18n.t('debtPreviewSaveButton')}
                    </Text>
                </Button>
            </Container>
        );
    }
}

const styles = {
    buttonStyle: {
        width: '90%',
        height: 40,
        backgroundColor: '#1c313a',
        marginTop: 20,
        marginBottom: 20,
        justifyContent: 'center',
        alignSelf: 'center'
    },
    itemStyle: {
        fontWeight: '600',
        alignSelf: 'center',
        color: 'white'
    }
};

const mapStateToProps = (state) => {
    const budgetID = state.budget.budgetID;
    const {name, amount, expirationDate, selectedCategories} = state.debt;
    return {name, amount, expirationDate, selectedCategories, budgetID};
};

const mapDispatchToProps = {
    createDebt
};

export default connect(mapStateToProps, mapDispatchToProps)(DebtPreview);
