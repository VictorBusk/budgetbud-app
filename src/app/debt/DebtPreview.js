import React, {Component} from 'react';
import {FlatList, View} from 'react-native';
import {
    Container,
    Content,
    Button,
    Spinner,
    ListItem,
    Body,
    Right,
    Text
} from "native-base";
import _ from 'lodash';
import {AppHeader, Separator} from "../../components/index";
import {connect} from "react-redux";
import {button, text, container, color} from "../../style/index";
import I18n from "../../strings/i18n";
import {showWarningToast} from "../../helpers/index";
import {createDebt, editDebt, getDebts, resetDebtError} from "./debt_actions";
import {getCategories} from "../category/category_actions";

class DebtPreview extends Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.debtError) {
            showWarningToast(nextProps.debtError);
            this.props.resetDebtError();
        }
    }

    onSavePress = async () => {
        if (this.props.debtLoading)
            return;

        if (this.props.selectedDebt)
            await this.props.editDebt(this.props);
        else
            await this.props.createDebt(this.props);

        this.props.navigation.pop(2);
        this.props.getDebts(this.props.budgetID);
        this.props.getCategories(this.props.budgetID);
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('debtPreviewHeader')}
                           showBackButton={true}
                           onLeftButtonPress={() => {
                               if (!this.props.debtLoading) {
                                   this.props.navigation.pop();
                               }
                           }}/>

                <Content style={{flex: 4}}>
                    <FlatList
                        data={this.props.categoryDebtItems}
                        renderItem={this.renderItem}
                    />
                </Content>

                <Separator/>

                <Button rounded
                        onPress={() => this.onSavePress()}
                        style={[button.defaultButton, color.button]}
                >
                    {this.props.debtLoading ? (
                        <Spinner color='#D0D0D0'/>) : (
                        <Text style={text.submitButtonText}>
                            {I18n.t('debtPreviewSaveButton')}
                        </Text>
                    )}
                </Button>
            </Container>
        );
    }

    renderItem = ({item}) => {
        return (
            <ListItem>
                <View style={container.fullWidth}>
                    <Body>
                    <Text style={color.text}>{item.name}</Text>
                    </Body>
                    <View style={[container.removeIndenting, {flexDirection: 'row'}]}>
                        <Body style={{flex: 1}}>
                        <Text note>{I18n.t('debtPreviewBefore')}</Text>
                        <Text note>{I18n.t('debtPreviewAfter')}</Text>
                        </Body>
                        <Right style={{flex: 2}}>
                            <Text note>{item.beforeAmount} {I18n.t('currency')}</Text>
                            <Text note>{item.afterAmount} {I18n.t('currency')} (-
                                {item.amountToSubtract} {I18n.t('currency')})
                            </Text>
                        </Right>
                    </View>
                </View>
            </ListItem>
        );
    }
}

const mapStateToProps = (state) => {
    const budgetID = state.budget.budgetID;
    const {
        name,
        totalAmount,
        expirationDate,
        categorySubtractions,
        selectedDebt,
        debtLoading,
        debtError
    } = state.debt;
    const {
        categories,
        categoriesOfDebt,
        selectedCategories
    } = state.category;

    const categoryDebtItems = _.map(selectedCategories, (item, key) => {
        const categorySubtraction = categorySubtractions.filter((obj) => {
            return obj.categoryID === item.toString();
        });

        const category = categories.filter((obj) => {
            return obj.categoryID === item.toString();
        });

        const amountToSubtract = categorySubtraction[0].amountToSubtract;
        let categoryOfDebtAmount = 0;

        if (selectedDebt) {
            const categoryOfDebt = categoriesOfDebt.filter((obj) => {
                return obj.categoryID === item.toString();
            });

            categoryOfDebtAmount = categoryOfDebt[0] ? categoryOfDebt[0].amount : 0;
        }

        const beforeAmount = category[0].amount + categoryOfDebtAmount;

        return {
            amountToSubtract: amountToSubtract,
            beforeAmount: beforeAmount,
            afterAmount: (beforeAmount - amountToSubtract),
            name: category[0].name,
            categoryID: item,
            key: key
        };
    });

    return {
        name,
        totalAmount,
        expirationDate,
        categoryDebtItems,
        selectedDebt,
        debtLoading,
        budgetID,
        debtError
    };
};

const mapDispatchToProps = {
    getDebts,
    createDebt,
    editDebt,
    getCategories,
    resetDebtError
};

export default connect(mapStateToProps, mapDispatchToProps)(DebtPreview);
