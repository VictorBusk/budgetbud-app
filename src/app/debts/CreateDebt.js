import React, {Component} from 'react';
import {connect} from "react-redux";
import {Container} from 'native-base';
import {AppHeader, DebtForm} from "../../components";
import I18n from "../../strings/i18n";
import {showWarningToast} from '../../helpers';
import {container} from "../../style";
import {
    amountChanged,
    calculateDebtCategorySubtractions,
    deleteDebt,
    expirationDateChanged,
    getDebts,
    nameChanged,
    resetDebtError,
    resetDebtForm,
    categoriesSelected,
    getCategories
} from "../../redux/actions";

class CreateDebt extends Component {
    componentWillMount() {
        this.props.resetDebtForm();
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.debtError) {
            showWarningToast(nextProps.debtError);
            this.props.resetDebtError();
        }
    }

    onContinuePress = () => {
        this.props.calculateDebtCategorySubtractions(
            this.props.name,
            this.props.totalAmount,
            this.props.expirationDate,
            this.props.selectedCategories, () => {
                this.props.navigation.navigate('DebtPreview');
            });
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('createDebtHeader')}
                           infoText={I18n.t('createDebtInfo')}
                           showBackButton={true}
                           onLeftButtonPress={() => {
                               if (!this.props.debtLoading) {
                                   this.props.navigation.pop();
                               }
                           }}/>

                <DebtForm nameChanged={this.props.nameChanged}
                          amountChanged={this.props.amountChanged}
                          expirationDateChanged={this.props.expirationDateChanged}
                          categoriesSelected={this.props.categoriesSelected}
                          name={this.props.name}
                          amount={this.props.totalAmount}
                          expirationDate={this.props.expirationDate}
                          categoryItems={this.props.categoryItems}
                          selectedCategories={this.props.selectedCategories}
                          categoriesLoading={this.props.categoriesLoading}
                          debtLoading={this.props.debtLoading}
                          onContinuePress={this.onContinuePress}/>
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    const {
        name,
        totalAmount,
        expirationDate,
        debtLoading,
        debtError
    } = state.debt;
    const {
        categories,
        categoriesOfDebt,
        selectedCategories,
        categoriesLoading,
    } = state.category;

    const categoryItems = categories.filter((obj) => {
        return obj.amount > 0
    });

    return {
        name,
        totalAmount,
        expirationDate,
        categoriesOfDebt,
        selectedCategories,
        categoryItems,
        categoriesLoading,
        debtLoading,
        debtError
    };
};

const mapDispatchToProps = {
    nameChanged,
    amountChanged,
    expirationDateChanged,
    calculateDebtCategorySubtractions,
    categoriesSelected,
    getCategories,
    getDebts,
    deleteDebt,
    resetDebtError,
    resetDebtForm
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateDebt);
