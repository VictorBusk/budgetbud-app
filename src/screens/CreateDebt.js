import React, {Component} from 'react';
import {connect} from "react-redux";
import {Container} from 'native-base';
import {AppHeader, DebtForm} from "../components/";
import _ from 'lodash';
import I18n from "../strings/i18n";
import {
    debtSelected,
    deleteDebt,
    getDebts,
    nameChanged,
    amountChanged,
    expirationDateChanged,
    categoriesOfDebtSelected,
    getCategories,
    calculateCategorySubtractions
} from "../actions";

class CreateDebt extends Component {
    componentWillMount() {
        this.props.getCategories(this.props.budgetID);
    }

    onContinuePress = () => {
        this.props.calculateCategorySubtractions(
            this.props.amount,
            this.props.expirationDate,
            this.props.categoriesOfDebtIDs, () => {
                this.props.navigation.navigate('DebtPreview');
            });
    };

    render() {
        return (
            <Container>
                <AppHeader headerText={I18n.t('createDebtHeader')}
                           showBackButton={true}
                           onLeftButtonPress={() => this.props.navigation.pop()}/>

                <DebtForm nameChanged={this.props.nameChanged}
                          amountChanged={this.props.amountChanged}
                          expirationDateChanged={this.props.expirationDateChanged}
                          categoriesSelected={this.props.categoriesOfDebtSelected}
                          name={this.props.name}
                          amount={this.props.amount}
                          expirationDate={this.props.expirationDate}
                          categoryItems={this.props.categoryItems}
                          selectedCategories={this.props.categoriesOfDebtIDs}
                          categoriesLoading={this.props.categoriesLoading}
                          subtractionsLoading={this.props.subtractionsLoading}
                          onContinuePress={this.onContinuePress}/>
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    const budgetID = state.budget.budgetID;
    const {name, amount, expirationDate} = state.debt;
    const {
        categories,
        categoriesOfDebtIDs,
        categoriesLoading,
        subtractionsLoading
    } = state.category;

    const categoriesWithAmounts = categories.filter((obj) => {
        return obj.categoryData.amount > 0
    });

    const categoryItems = _.map(categoriesWithAmounts, (item, key) => {
        return {...item.categoryData, categoryID: item.id, key: key};
    });

    return {
        name,
        amount,
        expirationDate,
        categoriesOfDebtIDs,
        budgetID,
        categoryItems,
        categoriesLoading,
        subtractionsLoading
    };
};

const mapDispatchToProps = {
    nameChanged,
    amountChanged,
    expirationDateChanged,
    calculateCategorySubtractions,
    categoriesOfDebtSelected,
    getCategories,
    debtSelected,
    getDebts,
    deleteDebt
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateDebt);
