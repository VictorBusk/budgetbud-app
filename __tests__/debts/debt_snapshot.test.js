import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CreateDebt from "../../src/app/debts/CreateDebt";
import {INITIAL_BUDGET_STATE, INITIAL_CATEGORY_STATE, INITIAL_DEBT_STATE} from "../test_helper/initial_state";
import thunk from 'redux-thunk';
import EditDebt from "../../src/app/debts/EditDebt";
import DebtOverview from "../../src/app/debts/DebtOverview";
import DebtPreview from "../../src/app/debts/DebtPreview";
import {DebtForm} from "../../src/components/forms/DebtForm";

Enzyme.configure({adapter: new Adapter()});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Debt', () => {
    it('renders CreateDebt as expected', () => {
        const wrapper = shallow(
            <CreateDebt/>,
            {
                context: {
                    store: mockStore({
                        debt: INITIAL_DEBT_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        budget: INITIAL_BUDGET_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders EditDebt as expected', () => {
        const wrapper = shallow(
            <EditDebt/>,
            {
                context: {
                    store: mockStore({
                        debt: INITIAL_DEBT_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        budget: INITIAL_BUDGET_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders DebtOverview as expected', () => {
        const wrapper = shallow(
            <DebtOverview/>,
            {
                context: {
                    store: mockStore({
                        debt: INITIAL_DEBT_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        budget: INITIAL_BUDGET_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders DebtPreview as expected', () => {
        const wrapper = shallow(
            <DebtPreview/>,
            {
                context: {
                    store: mockStore({
                        debt: INITIAL_DEBT_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        budget: INITIAL_BUDGET_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders DebtForm in loading state as expected', () => {
        const wrapper = shallow(
            <DebtForm debtLoading={true}/>
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});