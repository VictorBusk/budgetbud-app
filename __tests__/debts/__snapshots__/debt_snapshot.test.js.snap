// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debt renders CreateDebt as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Create Debt"
    infoText="Here you can create a debt item. Choose when the debt must be paid off and from which expense goals the debt should be subtracted. Press [Continue] to see a preview of the changes."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <DebtForm
    amount=""
    amountChanged={[Function]}
    categoriesLoading={false}
    categoriesSelected={[Function]}
    categoryItems={Array []}
    debtLoading={false}
    expirationDate=""
    expirationDateChanged={[Function]}
    name=""
    nameChanged={[Function]}
    onContinuePress={[Function]}
    selectedCategories={Array []}
  />
</Styled(Container)>
`;

exports[`Debt renders DebtForm in loading state as expected 1`] = `
<Styled(Container)
  accessible={false}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "90%",
        },
        Object {
          "paddingTop": 10,
        },
      ]
    }
  >
    <Styled(Label)
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "fontWeight": "400",
            "marginLeft": 5,
          },
          Object {
            "color": "#003755",
          },
        ]
      }
    >
      Enter name
    </Styled(Label)>
    <Styled(Item)
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "marginVertical": 5,
            "paddingVertical": 0,
          },
          Object {
            "backgroundColor": "#F6F7F9",
            "borderColor": "#003755",
          },
        ]
      }
    >
      <Styled(Input)
        onChangeText={[Function]}
        placeholder="Student loan"
        placeholderTextColor="#7F9BAA"
        style={
          Object {
            "color": "#003755",
          }
        }
      />
    </Styled(Item)>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 5,
        "marginTop": 10,
        "width": "90%",
      }
    }
  >
    <Styled(Label)
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "fontWeight": "400",
            "marginLeft": 5,
          },
          Object {
            "color": "#003755",
          },
        ]
      }
    >
      Enter amount
    </Styled(Label)>
    <Styled(Item)
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "marginVertical": 5,
            "paddingVertical": 0,
          },
          Object {
            "backgroundColor": "#F6F7F9",
            "borderColor": "#003755",
          },
        ]
      }
    >
      <Styled(Input)
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="5000"
        placeholderTextColor="#7F9BAA"
        style={
          Object {
            "color": "#003755",
          }
        }
        value="undefined"
      />
    </Styled(Item)>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Separator />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 5,
        "marginTop": 10,
        "width": "90%",
      }
    }
  >
    <Styled(Label)
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "fontWeight": "400",
            "marginLeft": 5,
          },
          Object {
            "color": "#003755",
          },
        ]
      }
    >
      Enter the desired date for the completed payment.
    </Styled(Label)>
    <DatePicker
      TouchableComponent={[Function]}
      androidMode="default"
      cancelBtnText="Cancel"
      confirmBtnText="Ok"
      customStyles={
        Object {
          "dateInput": Object {
            "backgroundColor": "#F6F7F9",
            "borderColor": "#003755",
            "borderRadius": 100,
            "borderWidth": 1,
            "marginRight": 5,
          },
          "dateText": Object {
            "color": "#003755",
            "fontSize": 17,
          },
        }
      }
      date=""
      disabled={false}
      duration={300}
      format="DD-MM-YYYY"
      height={259}
      hideText={false}
      iconComponent={
        <Styled(Icon)
          name="md-calendar"
          style={
            Object {
              "color": "#295E80",
            }
          }
          virtual={undefined}
        />
      }
      iconSource={1}
      minDate={2018-05-29T16:45:44.577Z}
      modalOnResponderTerminationRequest={[Function]}
      mode="date"
      onDateChange={[Function]}
      placeholder=""
      showIcon={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "marginVertical": 5,
            "paddingVertical": 0,
          },
          Object {
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 5,
      }
    }
  >
    <Separator />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <Styled(Label)
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "fontSize": 14,
              "fontWeight": "400",
              "marginLeft": 5,
            },
            Object {
              "color": "#003755",
            },
          ]
        }
      >
        Specify where the debt is to be deducted.
      </Styled(Label)>
    </View>
    <FlatList
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <Separator />
  </View>
  <Styled(Button)
    onPress={[Function]}
    rounded={true}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "width": "90%",
        },
        Object {
          "backgroundColor": "#00263A",
        },
      ]
    }
  >
    <Styled(Spinner)
      color="#D0D0D0"
    />
  </Styled(Button)>
</Styled(Container)>
`;

exports[`Debt renders DebtOverview as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <ConfirmDialog
    confirmCallback={[Function]}
    debtLoading={false}
    text="Do you want to delete the selected debt?"
    title="Confirm deletion"
  />
  <Styled(Container)>
    <AppHeader
      headerText="Debt overview"
      infoText="Here you can see all your created debt items. [Create debt] allows you to add a new debt item. Edit a debt item by pressing the pencil icon and delete the debt item by pressing the trash can icon."
      onLeftButtonPress={[Function]}
      showBackButton={true}
    />
    <Styled(Container)
      style={
        Object {
          "flex": 4,
          "justifyContent": "center",
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </Styled(Container)>
    <Separator />
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginTop": 20,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#00263A",
          },
        ]
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "600",
          }
        }
      >
        Create debt
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Container)>
</Styled(Container)>
`;

exports[`Debt renders DebtPreview as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Debt preview"
    infoText="Here is a preview of how your changes affect your budget. Press [Save] to save these changes."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <Styled(Content)
    style={
      Object {
        "flex": 4,
      }
    }
  >
    <FlatList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(Content)>
  <Separator />
  <Styled(Button)
    onPress={[Function]}
    rounded={true}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "width": "90%",
        },
        Object {
          "backgroundColor": "#00263A",
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontWeight": "600",
        }
      }
    >
      Save
    </Styled(Text)>
  </Styled(Button)>
</Styled(Container)>
`;

exports[`Debt renders EditDebt as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Edit debt"
    infoText="Here you can edit a debt item. Choose when the debt must be paid off and from which expense goals the debt should be subtracted. Press [Continue] to see a preview of the changes."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <DebtForm
    amount=""
    amountChanged={[Function]}
    categoriesLoading={false}
    categoriesSelected={[Function]}
    categoryItems={Array []}
    debtLoading={false}
    expirationDate=""
    expirationDateChanged={[Function]}
    name=""
    nameChanged={[Function]}
    onContinuePress={[Function]}
    selectedCategories={Array []}
  />
</Styled(Container)>
`;
